### DISCUSSION TITLE
Help with RandCropByPosNegLabeld

### DISCUSSION THREAD

**Discussion by Kiarashdnsh** [Project-MONAI/MONAI#6949] [Q&A] [answered]
Hello,

I'm a novice in AI and image segmentation.
I have a dataset of [256, 256, 256]-sized images and labels. The input channel is 1, and the output channel is 3 (label IDs are 0, 1, and 2, respectively). I'm trying to segment the images using UNet within the MONAI framework.

I've added the `RandCropByPosNegLabeld` transform to my input images since this random crop transform heavily raises the segmentation efficiency. However, in defining this transform, I've used the following code:
```python
RandCropByPosNegLabeld(
            keys=["image", "label"],
            label_key="label",
            spatial_size=(96, 96, 96),
            pos=1,
            neg=1,
            num_samples=4,
            image_key="image",
            image_threshold=0,
            ),
``` 
which dictates 4x(96, 96, 96) cropped images to be fed to the network as one image (as I've been reading [here](https://docs.monai.io/en/latest/transforms.html#randcropbyposneglabeld)); which results in 192 by 192 by 192 images.
My original size was [256, 256, 256]. By feeding [192, 192, 192] images to the network, how the network can accurately validate the original-sized images (I've not used this transform on my validation set)? Also, what has happened to the 64 remaining voxels that have been deleted during this crop? Does padding is used to retain the 256 image sizes?

Thanks in advance,
Kiarash

**Comment by KumoLiu** [comment, marked as answer]
Hi @Kiarashdnsh, you can use `SlidingWindowInferer` to infer and `Invert` to invert the previously applied transforms.
Here are some useful links:
https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d.ipynb
https://github.com/Project-MONAI/tutorials/blob/main/modules/inverse_transforms_and_test_time_augmentations.ipynb
Thanks!

**Reply by Kiarashdnsh** [reply]
From the examples, I understand that the `SlidingWindowInferer` uses the same ROIs to recreate the original images by reading consecutive cropped images, so in the validation phase, the network can segment original images with the same efficiency as the training phase. Also, the `Invert` returns the cropped images to the original images, in case validation on the original images is needed. Am I right?

**Reply by KumoLiu** [reply]
Yes, you are right. But `Invert` is mainly used to invert the label to help you visualize since you already have original images, so don't need to invert original images. :)

**Reply by Kiarashdnsh** [reply]
yeah, right! Thanks for your input and your help!
